### PR TITLE
Need to preserve the Xattr security.capability when chowning.

### PR DESCRIPTION
**Pull request by rhatdan**
When we chown we not only loose the setuid flags but we also loose
the file capabilties.  This means in a user namespace when we chown
ping, it looses its filecap, which means it will not longer work.

This fix will check for the security capabilies and retain them if they
exist when chowning.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@nalind @giuseppe PTAL

File: drivers/chown_unix.go [hunk 1]
```diff
@@ -49,6 +50,11 @@ func platformLChown(path string, info os.FileInfo, toHost, toContainer *idtools.
 			if err != nil {
 				return fmt.Errorf("%s: lstat(%q): %v", os.Args[0], path, err)
 			}
+			cap, err := system.Lgetxattr(path, "security.capablity")
```
Comment (vrothberg) [inline drivers/chown_unix.go]: typo s/capablity/capability/ ("i" missing)

**Comment by TomSweeneyRedHat** [comment]
Ditto what @vrothberg said

File: drivers/chown_unix.go [hunk 2]
```diff
@@ -49,6 +50,11 @@ func platformLChown(path string, info os.FileInfo, toHost, toContainer *idtools.
 			if err != nil {
 				return fmt.Errorf("%s: lstat(%q): %v", os.Args[0], path, err)
 			}
+			cap, err := system.Lgetxattr(path, "security.capability")
+			if err != nil {
+				return fmt.Errorf("%s: Lgetxattr(%q): %v", os.Args[0], path, err)
```
Comment (nalind) [inline drivers/chown_unix.go]: AIUI this will always return an error on non-Linux hosts.

Comment (nalind) [reply drivers/chown_unix.go]: ... so we either need to check for and ignore `system.ErrNotSupportedPlatform`, or add a note that we're going to have to fix this later.

**Comment by rhatdan** [comment]
@vrothberg Yes I will add a test to podman for this once it gets merged.

**Comment by giuseppe** [comment]
LGTM

**Comment by nalind** [comment]
LGTM

**Pull request merged**